substrate/client/network/src/peer_store.rs

```diff
@@ -95,14 +107,11 @@ pub trait PeerStoreProvider: Debug + Send + Sync {
 		ignored: HashSet<sc_network_types::PeerId>,
 	) -> Vec<sc_network_types::PeerId>;
 
-	/// Get the number of known peers.
-	///
-	/// This number might not include some connected peers in rare cases when their reputation
-	/// was not updated for one hour, because their entries in [`PeerStore`] were dropped.
-	fn num_known_peers(&self) -> usize;
-
 	/// Add known peer.
 	fn add_known_peer(&self, peer_id: sc_network_types::PeerId);
+
+	/// Get the status of the peer store.
+	fn status(&self) -> PeerStoreStatus;
```

Comment:
Looking at the implementation of this function. Why don't we update the metrics directly in the peerstore? Then we don't need to lock any mutexes?